graphics/lvgl/Makefile

```diff
@@ -78,20 +78,21 @@ $(LVGL_UNPACKNAME): $(LVGL_TARBALL)
 	$(Q) mv	lvgl-$(LVGL_VERSION) $(LVGL_UNPACKNAME)
 	$(Q) touch $(LVGL_UNPACKNAME)
 
-lvgl/lvgl.h: $(LVGL_UNPACKNAME)
+$(LVGL_UNPACKNAME)/lvgl.h: $(LVGL_UNPACKNAME)
 
-$(APPDIR)/include/graphics/lvgl.h: lvgl/lvgl.h
-	@echo "CP: lvgl/lvgl.h"
-	$(Q) cp lvgl/lvgl.h $(APPDIR)/include/graphics/lvgl.h
+exports/graphics/lvgl.h: $(LVGL_UNPACKNAME)/lvgl.h
+	$(Q) mkdir -p exports/graphics
+	@echo "CP: $(LVGL_UNPACKNAME)/lvgl.h"
+	$(Q) cp $(LVGL_UNPACKNAME)/lvgl.h exports/graphics/lvgl.h
```

Comment:
Why copy lvgl.h to exports directory? How about we export?
CFLAGS += ${shell $(INCDIR) $(INCDIROPT) "$(CC)" $(APPDIR)/graphics/lvgl}
CXXFLAGS += ${shell $(INCDIR) $(INCDIROPT) "$(CC)" $(APPDIR)/graphics/lvgl}

Reply:
Sure, but then we need to `#include <lvgl.h>` instead of `#include <graphics/lvgl.h>`. I don't mind but I though that this prefix is intentional to keep things within namespaces like `graphics` here. Do we want keep prefixes or do they go? So far they were used consistently.

Here is additional PR which shows necessary change to introduce this suggestion https://github.com/w8jcik/incubator-nuttx-apps/pull/1/files

Reply:
Update:

Acutally it will be `#include <lvgl/lvgl.h>` not <lvgl.h>, there was one more include present in `lvgldemo` which I just noticed.

https://github.com/w8jcik/incubator-nuttx-apps/pull/1/commits/7f62736c771922aa8c8e6bd3d4976d083ef66ef8

But this is LittlevGL specific. The question remains, what to do with prefixes.

Reply:
Usually libraries assume some sort of layout for the include directory and the user only needs to account for installation prefix. In this case, lvgl is assuming a layout of <prefix>/lvgl/lvgl.h, so any extra component in the path should be part of the -I flag passed to the compiler. Adding extra prefixes could break existing programs following the expected inclusion format.

Reply:
We should follow LVGL convention:
https://github.com/lvgl/lv_examples/blob/master/lv_examples.h#L16
either ```#include <lvgl.h>``` or ```#include <lvgl/lvgl.h> is good, but ```#include <graphics/lvgl.h>``` just make the life of programmer is hard.

Reply:
I think "#include <lvgl/lvgl.h>" as Maciej modified is fine. Currently only "lvgldemo" is using the LVGL. Maciej, avoid using the name LittlevGL because it is "deprecated", see: https://blog.lvgl.io/2020-06-01/announcement

Reply:
Hi @v01d the prefix is already present, because currently littlevgl is copying some header to `graphics`. Change suggested by @xiaoxiang781216 removes this prefix. I was asking should we really remove it, because this way we break compatibility. Breaking compatibility is fine for me. But what is the official strategy. Do we want `graphics` in front of all lvgl includes? Right now it seems to be in front of one of the headers and probably not in front of the others. I think it should be still possible to have `#include <graphics/lvgl/...>`. The question is do we want it.